integration/kubernetes/k8s-inotify.bats

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bats
+#
+# Copyright (c) 2021 Apple Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/../../.ci/lib.sh"
+load "${BATS_TEST_DIRNAME}/tests_common.sh"
+
+setup() {
+	[ "${KATA_HYPERVISOR}" == "firecracker" ] && skip "test not working see: ${fc_limitations}"
+	export KUBECONFIG="${KUBECONFIG:-$HOME/.kube/config}"
+	get_pod_config_dir
+}
+
+@test "configmap update works, and preserves symlinks" {
+	[ "${KATA_HYPERVISOR}" == "firecracker" ] && skip "test not working see: ${fc_limitations}"
```

Comment:
Hi @egernst ,

Always bothered me see the duplication of skip instructions on each tests. In my mind if `setup` skips then the tests should automatically be skipped as well. Then I gave it a try:
```
[wmoschet@wainer-laptop tmp]$ cat foo.bats 
setup() {
	skip "skip setup"
	echo "run setup"
}

@test "test1" {
	skip "skip test1"
}

@test "test2" {
	echo "run test2"
}
[wmoschet@wainer-laptop tmp]$ bats foo.bats 
 - test1 (skipped: skip setup)
 - test2 (skipped: skip setup)

2 tests, 0 failures, 2 skipped

[wmoschet@wainer-laptop tmp]$ bats --version
Bats 1.3.0
```
At least with bats 1.3.0 having `skip` on `setup()` is enough. However I don't know about older versions. We don't set a bats version on any `versions.yaml` (and `tests/.ci/install_bats.sh` pull and install the latest code).

Then I thought "what about the teardown()?". I gave it a try again:
```
[wmoschet@wainer-laptop tmp]$ cat foo2.bats 
setup() {
	skip "skip setup"
	echo "run setup"
}

@test "test1" {
	skip "skip test1"
}

@test "test2" {
	echo "run test2"
}

teardown() {
	/bin/false
}
[wmoschet@wainer-laptop tmp]$ bats foo2.bats 
 ✗ test1
   (from function `teardown' in test file foo2.bats, line 15)
     `/bin/false' failed
 ✗ test2
   (from function `teardown' in test file foo2.bats, line 15)
     `/bin/false' failed

2 tests, 2 failures

```
Yeah, it needs to skip `teardown()` too:
```
[wmoschet@wainer-laptop tmp]$ cat foo3.bats
setup() {
	skip "skip setup"
	echo "run setup"
}

@test "test1" {
	skip "skip test1"
}

@test "test2" {
	echo "run test2"
}

teardown() {
	skip "skip teardown"
}
[wmoschet@wainer-laptop tmp]$ bats foo3.bats 
 - test1 (skipped: skip setup)
 - test2 (skipped: skip setup)

2 tests, 0 failures, 2 skipped
```

Reply:
Nice -- good catch!  I'll update the PR to (1) include more tests, (2) skip the in-test skipping!